thirdeye/thirdeye-frontend/app/pods/components/rootcause-metrics/component.js

```diff
@@ -27,11 +27,57 @@ export default Ember.Component.extend({
    */
   isLoading: false,
 
+  /**
+   * Columns for metrics table
+   * @type Object[]
+   */
+  metricsTableColumns: [
+    {
+      template: 'custom/table-checkbox'
+    }, {
+      propertyName: 'metric',
+      title: 'Metric Name',
+      className: 'rootcause-metric__table__column'
+    }, {
+      propertyName: 'score',
+      title: 'Anomalous Score',
+      className: 'rootcause-metric__table__column'
+    }, {
+      template: 'custom/metrics-table-changes',
+      propertyName: 'change',
+      title: 'Changes',
+      className: 'rootcause-metric__table__column'
+    }
+  ],
+
   init() {
     this._super(...arguments);
     this.setProperties({ sortProperty: ROOTCAUSE_METRICS_SORT_PROPERTY_CHANGE, sortMode: ROOTCAUSE_METRICS_SORT_MODE_ASC });
   },
 
+  /**
+   * Data for metrics table
+   * @type Object[] - array of objects, each corresponding to a row in the table
+   */
+  metricsTableData: Ember.computed(
+    'urns',
```

Comment:
Are we expecting the value of these properties to change together or can each one be changed individually while the others remain the same? Based on the function, it seems like all these values are updated at once together. If so, it would seem like making a computed property with observers for each of these properties creates unnecessary references. If not, ignore this comment :)

Reply:
Yeah, I notice that too. The engineer who wrote the logic for computing those properties are out, so I'll have to ask him for more context. Until then, I figured observing for all these changes is a safer approach. Thanks for the comment.